pyintegration/integration-tests/numbers_from_list_and_func.py

```diff
@@ -0,0 +1,37 @@
+#
+#   Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+from deephaven import dtypes
+from deephaven import empty_table
+from deephaven.constants import MAX_BYTE, MAX_SHORT, MAX_INT, MAX_LONG
+
+x = [MAX_BYTE, MAX_SHORT, MAX_INT, MAX_LONG, 0.98888, 999999.888888]
+n = len(x)
+
+
+def get_x(i):
+    return x[i]
+
+
+t_list = empty_table(n).update(["X = x[i]"])
+t_func = empty_table(n).update(["X = get_x(i)"])
+assert t_list.columns[0].data_type == dtypes.PyObject
+assert t_func.columns[0].data_type == dtypes.JObject
```

Comment:
personally i dont care for these assertions except as a sort of "hey this changed", but we dont actually require this difference, right?

it isn't bad to have the assertions, but maybe add a note that this behavior isnt actually essential, it is just what happens today or something?